api/src/main/java/com/datatorrent/api/Attribute.java

```diff
@@ -146,6 +146,8 @@ public String toString()
 
     Set<Map.Entry<Attribute<?>, Object>> entrySet();
 
+    <T> T getValue(Attribute<T> key);
```

Comment:
It's a commonly needed functionality as we already have the notion of the default attribute value for an Attribute and the older method doesn't take it into account. The default value is being used for the launch parameters.